primitives/runtime/src/generic/header.rs

```diff
@@ -43,6 +44,7 @@ pub struct Header<Number: Copy + Into<U256> + TryFrom<U256>, Hash: HashT> {
 		feature = "std",
 		serde(serialize_with = "serialize_number", deserialize_with = "deserialize_number")
 	)]
+	#[codec(compact)]
```

Comment:
Is this intended?

Reply:
Indeed, see https://github.com/paritytech/substrate/pull/8615#discussion_r680975235.

I'm not 100% sure why the `Encode` cannot also be derived.

Reply:
Oh ok, so I was initially worried about the encoding representation of the header number, but I see now that `Header` doesn't even derive Encode or Decode at all, and this here is purely used for deriving `TypeInfo`. All good for me.

Reply:
imo using compact here to derive typeinfo feels like a hack, but 🤷‍♂️

Reply:
It is indeed a hack, but IMO less error prone than the manual impl https://github.com/paritytech/substrate/pull/8615/commits/9d0b2170adf80c73f8f086425cb29f06681859c6

Reply:
Why don't we use the macro derive here for `Encode` and `Decode`? Looking at the manual implementation, I don't see any reason?

Reply:
Resolved in #9489